pallets/did/src/utils.rs

```diff
@@ -98,6 +79,33 @@ fn check_is_valid_ascii_string() {
 	});
 }
 
+#[test]
+fn check_is_valid_uri() {
+	let test_cases = [
+		("kilt.io", true),
+		("super.long.domain.com:12345/path/to/directory#fragment?arg=value", true),
+		("super.long.domain.com:12345/path/to/directory/file.txt", true),
+		("domain.with.only.valid.characters.:/?#[]@!$&'()*+,;=-._~", false),
```

Comment:
Why is this URI wrong? Maybe a comment for the false cases can help to quicker understand.

Reply:
Clearer like this? 😊 https://github.com/KILTprotocol/kilt-node/pull/545/commits/cc9c79146fd26e206e2bbda64f4ff74f62a5557e